### ISSUE TITLE
default spacing is really too big

### ISSUE THREAD

**Issue by rcoreilly** [cogentcore/core#909]
### Describe the bug

maybe have that be the default on mobile, but on desktop, it is just crazy spaced for a default.

### How to reproduce

run anything.  expt with spacing option in settings.

### Platform

macOS

**Comment by ddkwork** [comment]
I have same problem

**Comment by kkoreilly** [comment]
This is simply not true. Our current default spacing is good for user-friendly applications and is consistent with that of Material Design. The user is always free to set the spacing to something else.

**Comment by ddkwork** [comment]
I have same thing![{F7B5CD03-0025-47fd-9384-01A955CCA7CB}.png](https://github.com/cogentcore/core/assets/19886504/115f588c-5f71-4262-99ef-45dc6924599c)

not working

**Comment by kkoreilly** [comment]
The appearance settings zoom and spacing are based around 0-100, not 0-1, so those should be 50 and 70, respectively. However, again, I would recommend against hardcoding the settings in your code, as that prevents people from changing them. I will look into figuring out better default spacing settings, but you should not force your opinion in your app; on some computers, you need a much bigger or smaller zoom. Once you set your settings in the app, they will be saved and persist for all apps on that device.

**Comment by ddkwork** [comment]
Again, I feel like you're pointing out the maze, it's strange that my Android phone and tablet are very suitable for this ratio, by the way, I also have an all-in-one computer and another laptop next to me, I'll compile exe to them later for testing, if the ratio is still appropriate, then I'm here for 4 devices that are suitable for this ratio, and I'll report back to you after I test, after all, I can't see the less tested.



---Original---
From: ***@***.***&gt;
Date: Thu, Feb 29, 2024 23:43 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [cogentcore/core] default spacing is really too big (Issue #909)




 
The appearance settings zoom and spacing are based around 0-100, not 0-1, so those should be 50 and 70, respectively. However, again, I would recommend against hardcoding the settings in your code, as that prevents people from changing them. I will look into figuring out better default spacing settings, but you should not force your opinion in your app; on some computers, you need a much bigger or smaller zoom. Once you set your settings in the app, they will be saved and persist for all apps on that device.

**Comment by ddkwork** [comment]
The main thing is that the default zoom, the phone and tablet are too big



---Original---
From: ***@***.***&gt;
Date: Thu, Feb 29, 2024 23:43 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [cogentcore/core] default spacing is really too big (Issue #909)




 
The appearance settings zoom and spacing are based around 0-100, not 0-1, so those should be 50 and 70, respectively. However, again, I would recommend against hardcoding the settings in your code, as that prevents people from changing them. I will look into figuring out better default spacing settings, but you should not force your opinion in your app; on some computers, you need a much bigger or smaller zoom. Once you set your settings in the app, they will be saved and persist for all apps on that device.

**Comment by ddkwork** [comment]
![{6AD7D576-EA1C-4adb-AC57-D3AE42514331}.png](https://github.com/cogentcore/core/assets/19886504/0c2b7f99-7b71-4196-87f8-89882dc77cf6)

**Comment by ddkwork** [comment]
Unfortunately, I can't install the VK driver for the other two devices, they are old devices and don't support VK, so I can't test it, so let's do it first, regardless of this problem, just adjust it manually



---Original---
From: ***@***.***&gt;
Date: Thu, Feb 29, 2024 23:43 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [cogentcore/core] default spacing is really too big (Issue #909)




 
The appearance settings zoom and spacing are based around 0-100, not 0-1, so those should be 50 and 70, respectively. However, again, I would recommend against hardcoding the settings in your code, as that prevents people from changing them. I will look into figuring out better default spacing settings, but you should not force your opinion in your app; on some computers, you need a much bigger or smaller zoom. Once you set your settings in the app, they will be saved and persist for all apps on that device.

**Comment by kkoreilly** [comment]
Did you restart the app after you changed the system scaling factor? Do you have a different font size specified in your accessibility settings? Can you send me a screenshot of `Inspect` > `Inspect app` > `Screens` > right click > `Edit`?

**Comment by ddkwork** [comment]
After the zoom in the system settings is adjusted to 100%, the whole system is quite small, the mouse is difficult to step text, start menu, explorer and the like, it is not easy to operate, this can be a low priority, and when I find out that there is a computer that supports VK driver, I will further test it and report the results to you



---Original---
From: ***@***.***&gt;
Date: Fri, Mar 1, 2024 04:33 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [cogentcore/core] default spacing is really too big (Issue #909)




 
Did you restart the app after you changed the system scaling factor? Do you have a different font size specified in your accessibility settings? Can you send me a screenshot of Inspect &gt; Inspect app &gt; Screens &gt; right click &gt; Edit?

**Comment by ddkwork** [comment]
> Did you restart the app after you changed the system scaling factor? Do you have a different font size specified in your accessibility settings? Can you send me a screenshot of `Inspect` > `Inspect app` > `Screens` > right click > `Edit`?

is this ?
<img width="1280" alt="{4324D9EA-60FE-432c-BBBE-11A2E6D455F3}" src="https://github.com/cogentcore/core/assets/19886504/476e6af4-9a8f-4df7-91b5-65c922100d36">

**Comment by kkoreilly** [comment]
Yes. It says that you have a `Device pixel ratio` of 1.5, which means that it thinks your system has 150% zoom. Can you show me a screenshot of that same page with your system widgets (taskbar, etc) included, with the system zoom and Cogent Core zoom at 100% and the app restarted to show the difference between the system and Cogent Core?

**Comment by ddkwork** [comment]
2762713521
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "cogentcore/core"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月1日(星期五) 上午10:47
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [cogentcore/core] default spacing is really too big (Issue #909)





 
Yes. It says that you have a Device pixel ratio of 1.5, which means that it thinks your system has 150% zoom. Can you show me a screenshot of that same page with your system widgets (taskbar, etc) included, with the system zoom and Cogent Core zoom at 100% and the app restarted to show the difference between the system and Cogent Core?

**Comment by kkoreilly** [comment]
It says that it is an invalid address.

**Comment by ddkwork** [comment]
wait me，sorry

**Comment by ddkwork** [comment]
![{14913537-D06F-49b3-8B97-E2D454E38A34}.png](https://github.com/cogentcore/core/assets/19886504/796ffa26-f8cc-49e9-a1ac-86dded1d9ab9)

![{0A6902E6-3588-4a70-9030-00CC4BB619BD}.png](https://github.com/cogentcore/core/assets/19886504/90acd379-89d9-4333-9f69-ec56d7c7cfb4)

**Comment by kkoreilly** [comment]
For me, the Cogent Core file view is only slightly bigger than the Windows one. Can you send a screenshot of the same thing (the Cogent Core file view alongside the Windows one). Also, are you sure your zoom and spacing are set to 100 in the Cogent Core settings?

![image](https://github.com/cogentcore/core/assets/37386857/4704d695-22bd-4f16-8962-30ef3de101f2)

**Comment by ddkwork** [comment]
![{EC9985DF-039C-4fe3-A59F-448A9AF66EB8}.png](https://github.com/cogentcore/core/assets/19886504/fb77f088-1114-444c-969b-6a3f28a2d139)

**Comment by ddkwork** [comment]
```go
//this looks like well to me

func Test_well(t *testing.T) {
	zoom, spacing := well(150)
	assert.Equal(t, float64(50), zoom)
	assert.Equal(t, float64(70), spacing)

	zoom, spacing = well(100)
	assert.Equal(t, float64(0), zoom)
	assert.Equal(t, float64(20), spacing)
}

func well(systemScale float64) (zoom, spacing float64) {
	return systemScale - 100, systemScale - 80
}

```
